pkgs/development/tools/build-managers/ninja/setup-hook.sh

```diff
@@ -1,26 +1,21 @@
 ninjaBuildPhase() {
     runHook preBuild
 
-    if [[ -z "$ninjaFlags" && ! ( -e build.ninja ) ]]; then
-        echo "no build.ninja, doing nothing"
-    else
-        local buildCores=1
+    local buildCores=1
 
-        # Parallel building is enabled by default.
-        if [ "${enableParallelBuilding-1}" ]; then
-            buildCores="$NIX_BUILD_CORES"
-        fi
+    # Parallel building is enabled by default.
+    if [ "${enableParallelBuilding-1}" ]; then
+        buildCores="$NIX_BUILD_CORES"
+    fi
 
-        # shellcheck disable=SC2086
-        local flagsArray=( \
-            -j"$buildCores" -l"$NIX_BUILD_CORES" \
-            $ninjaFlags "${ninjaFlagsArray[@]}" \
-            $buildFlags "${buildFlagsArray[@]}")
+    local flagsArray=(
+        -j$buildCores -l$NIX_BUILD_CORES
+        $ninjaFlags "${ninjaFlagsArray[@]}"
+    )
 
-        echoCmd 'build flags' "${flagsArray[@]}"
-        ninja "${flagsArray[@]}"
-        unset flagsArray
-    fi
+    echoCmd 'build flags' "${flagsArray[@]}"
+    ninja "${flagsArray[@]}"
+    unset flagsArray
```

Comment:
I have just noticed: since `flagsArray` is local, we don't have to unset it.

Reply:
I noticed that too but thought it might be to avoid it leaking into the post hook?

Reply:
`runHook` is a function, it has a new scope without local variables of the current function.